packages/react-components/react-avatar/src/components/AvatarGroup/useAvatarGroupStyles.ts

```diff
@@ -18,34 +22,179 @@ const useStyles = makeStyles({
+  thickest: { [avatarGroupItemOutlineVar]: tokens.strokeWidthThickest },
+  xxs: { [avatarGroupItemMarginVar]: `calc(-1 * ${tokens.spacingHorizontalXXS})` },
+  xs: { [avatarGroupItemMarginVar]: `calc(-1 * ${tokens.spacingHorizontalXS})` },
+  s: { [avatarGroupItemMarginVar]: `calc(-1 * ${tokens.spacingHorizontalS})` },
+  l: { [avatarGroupItemMarginVar]: `calc(-1 * ${tokens.spacingHorizontalL})` },
 });
 
+const useSpreadStyles = makeStyles({
+  s: { [avatarGroupItemMarginVar]: tokens.spacingHorizontalS },
+  mNudge: { [avatarGroupItemMarginVar]: tokens.spacingHorizontalMNudge },
+  m: { [avatarGroupItemMarginVar]: tokens.spacingHorizontalM },
+  l: { [avatarGroupItemMarginVar]: tokens.spacingHorizontalL },
+  xl: { [avatarGroupItemMarginVar]: tokens.spacingHorizontalXL },
+});
+
+/**
+ * Styles for overflow list slot.
+ */
 const useOverflowListStyles = makeStyles({
   base: {
     listStyleType: 'none',
-    ...shorthands.margin(0),
-    ...shorthands.padding(0),
+    maxHeight: '220px',
+    minHeight: '80px',
+    ...shorthands.overflow('hidden', 'scroll'),
+    ...shorthands.padding(tokens.spacingHorizontalS),
+    width: '220px',
   },
 });
 
 /**
  * Apply styling to the AvatarGroup slots based on the state
  */
 export const useAvatarGroupStyles_unstable = (state: AvatarGroupState): AvatarGroupState => {
-  const { size } = state;
+  const { layout, overflowIndicator, size } = state;
   const styles = useStyles();
   const sizeStyles = useSizeStyles();
   const overflowListStyles = useOverflowListStyles();
+  const stackStyles = useStackStyles();
+  const spreadStyles = useSpreadStyles();
+  const overflowButtonStyles = useOverflowButtonStyles();
 
-  state.root.className = mergeClasses(avatarGroupClassNames.root, styles.base, state.root.className);
+  const rootClasses = [];
 
-  if (state.overflowSurface) {
-    state.overflowSurface.className = mergeClasses(
-      avatarGroupClassNames.overflowSurface,
-      state.overflowSurface.className,
-    );
+  if (layout === 'stack') {
+    if (size < 56) {
+      rootClasses.push(stackStyles.thick);
+    } else if (size < 72) {
+      rootClasses.push(stackStyles.thicker);
+    } else {
+      rootClasses.push(stackStyles.thickest);
+    }
+
+    if (size < 24) {
+      rootClasses.push(stackStyles.xxs);
+    } else if (size < 48) {
+      rootClasses.push(stackStyles.xs);
+    } else if (size < 96) {
+      rootClasses.push(stackStyles.s);
+    } else {
+      rootClasses.push(stackStyles.l);
+    }
+  } else if (layout === 'spread') {
+    if (size < 20) {
+      rootClasses.push(spreadStyles.s);
+    } else if (size < 32) {
+      rootClasses.push(spreadStyles.mNudge);
+    } else if (size < 64) {
+      rootClasses.push(spreadStyles.l);
+    } else {
+      rootClasses.push(spreadStyles.xl);
+    }
   }
 
+  state.root.className = mergeClasses(
+    avatarGroupClassNames.root,
+    styles.base,
+    layout === 'pie' && styles.pie,
+    layout === 'pie' && sizeStyles[size],
```

Comment:
I thought maybe they should stay there since the if statements take much mor space and these are a single line each. I'll add them the same way as the spacing styles for consistency!